benchmark/syclblas/blas1/asum.cpp

```diff
@@ -44,24 +44,26 @@ void run(benchmark::State& state, ExecutorType* executorPtr, index_t size,
 
   ExecutorType& ex = *executorPtr;
 
+  using data_t = utils::data_storage_t<scalar_t>;
+
   // Create data
-  std::vector<scalar_t> v1 = blas_benchmark::utils::random_data<scalar_t>(size);
-  scalar_t vr;
+  std::vector<data_t> v1 = blas_benchmark::utils::random_data<data_t>(size);
+  data_t vr;
 
-  auto inx = blas::make_sycl_iterator_buffer<scalar_t>(v1, size);
```

Comment:
A buffer iterator can create a subset of the buffer/data and the data passed as an input is not necessarily going to be a vector. Actually, it is a raw pointer, for most cases. Removing this can break the c-style support of BLAS, which should be traditionally maintained. So the size must be passed as an input.

Reply:
This is just removing some calls from the tests and benchmarks, and replacing them with utility functions that calls this internally. Additionally there are still a few places where this is used.

Reply:
If that's the case and you can still make a subset manually then that's fine I think.